perfmetrics/scripts/bigquery/bigquery.py

```diff
@@ -0,0 +1,317 @@
+      config_name (str): Name of the experiment configuration.
+
+    Returns:
+      str: Configuration ID of the experiment
+    """
+    # Check if the experiment configuration is already present in table
+    query_check_config_exists = """
+      SELECT configuration_id
+      FROM `{}.{}.{}`
+      WHERE gcsfuse_flags = '{}'
+      AND branch = '{}'
+      AND configuration_name = '{}'
+#     """.format(self.project_id, self.dataset_id, ExperimentsGCSFuse.CONFIGURATION_TABLE_ID, gcsfuse_flags, branch, config_name)
+
+    job = self._execute_query_and_check_for_error(query_check_config_exists)
+    result_count = job.result().total_rows
+
+    # If result empty, then experiment configuration not present -> insert new experiment configuration -> return configuration ID
+    if result_count == 0:
+      ds_ref = self.dataset_ref
+      table_ref = ds_ref.table(ExperimentsGCSFuse.CONFIGURATION_TABLE_ID)
+      table = self.client.get_table(table_ref)
+      uuid_str = str(uuid.uuid4())
+      rows_to_insert = [(uuid_str, config_name, gcsfuse_flags, branch, end_date)]
+      result = self.client.insert_rows(table, rows_to_insert)
+      if result:
+        print(f'Error inserting experiment configuration: {result}')
+        sys.exit(1)
+      return uuid_str
+
+    # If more than 1 result -> duplicate experiment configuration present -> throw error
+    elif result_count > 1:
+      print("Duplicate experiment configurations exist. Data corrupted")
+      sys.exit(1)
+
+    # If exactly one result -> update end date -> return configuration ID
+    else:
+      config_id = None
+      for row in job:
+        config_id = row['configuration_id']
+      query_update_end_date = """
+        UPDATE `{}.{}.{}`
+        SET end_date = '{}'
+        WHERE configuration_id = '{}'
+        """.format(self.project_id, self.dataset_id, ExperimentsGCSFuse.CONFIGURATION_TABLE_ID, end_date, config_id)
+      self._execute_query_and_check_for_error(query_update_end_date)
+      return config_id
+
+  def upload_metrics_to_table(self, table_name, config_id, start_time_build, metrics_data):
+
+    """Uploads metrics_data to the table corresponding to 'table_name'.
+
+    Args:
+      table_name (str): Table name corresponding to the table id to which results are being uploaded
+      config_id (str): config_id of the experiment for which results are being uploaded
+      start_time_build (timestamp): Start time of the build
+      metrics_data (list): A 2D list containing the experiment results
+    """
+
+    # Check if the configuration ID of the experiment is valid
+    config_valid = self._check_if_config_valid(config_id)
+
+    if not config_valid:
+      print("Invalid configuration ID")
+      sys.exit(1)
+    else:
+      # Get the dataset reference
+      ds_ref = self.dataset_ref
+      table_id = None
+      # Get the table ID based on the table name
+      if table_name == 'fio':
```

Comment:
how about maintaining a dictionary from name to id mapping and then getting/checking the key in that dict here ?